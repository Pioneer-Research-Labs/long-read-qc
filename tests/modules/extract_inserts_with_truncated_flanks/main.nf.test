nextflow_process {

    name "Test Process extract_inserts_with_truncated_flanks"
    script "modules/extract_inserts_with_truncated_flanks/main.nf"
    process "extract_inserts_with_truncated_flanks"

    test("Should find inserts from untrimmed fastq file") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id: 'HE_USER-gapfill_Lib1_20241220',
                genome:'H_elongata'],
                file('${projectDir}/test_data/HE_USER-gapfill_Lib1_20241220/flanking.gb'),
                file('${projectDir}/test_data/HE_USER-gapfill_Lib1_20241220/untrimmed.fastq'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
            ("$outputDir/HE_USER-gapfill_Lib1_20241220/inserts_from_truncated_flanks.fasta"),
            ("$outputDir/HE_USER-gapfill_Lib1_20241220/cutadapt_inserts_report_from_truncated_flanks.json"),
            ("$outputDir/HE_USER-gapfill_Lib1_20241220/untrimmed_from_truncated_flanks.fastq"),
            ("$outputDir/HE_USER-gapfill_Lib1_20241220/cutadapt_info_from_truncated_flanks.tsv")).match()
        }

    }

}
