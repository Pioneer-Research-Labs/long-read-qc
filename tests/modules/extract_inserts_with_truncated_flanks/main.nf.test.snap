{
    "Should find inserts from untrimmed fastq file": {
        "content": [
            "inserts_from_truncated_flanks.fasta:md5,65bd0e2fc22cfa8a915a8d5219d6d2e5",
            "cutadapt_inserts_report_from_truncated_flanks.json:md5,5c87b04b5964a652ff8e307315e47e56",
            "untrimmed_from_truncated_flanks.fastq:md5,40b4067bc8883a4cf1f664c1ab2e0be1",
            "cutadapt_info_from_truncated_flanks.tsv:md5,45a4c622ce3d6459bc6e9c2bb5fcbd27"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-30T16:15:47.598925"
    }
}